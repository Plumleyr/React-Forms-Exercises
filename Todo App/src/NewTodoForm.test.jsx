import {render, fireEvent, screen} from '@testing-library/react'
import NewTodoForm from './NewTodoForm'
import { expect, it, vi } from 'vitest'

it('should render without crashing', () => {
  render(<NewTodoForm/>)
})

it('should match snapshot', () => {
  const {asFragment} = render(<NewTodoForm/>);
  expect(asFragment()).toMatchSnapshot();
})

it('should call addBox with appropriate information', () => {
  const addTodoMock = vi.fn()

  render(<NewTodoForm addTodo={addTodoMock}/>)

  fireEvent.change(screen.getByPlaceholderText('todo'), {
    target: {value: 'take out trash'}
  });

  fireEvent.click(screen.getByText('Create Todo'))

  expect(addTodoMock).toHaveBeenCalledTimes(1);
  expect(addTodoMock).toHaveBeenCalledWith({
    todo: 'take out trash'
  });
});