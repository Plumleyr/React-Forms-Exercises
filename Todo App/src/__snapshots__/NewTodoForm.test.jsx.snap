// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo: 
    </label>
    <input
      id="todo"
      name="todo"
      placeholder="todo"
      type="text"
      value=""
    />
    <button>
      Create Todo
    </button>
  </form>
</DocumentFragment>
`;
